docs/data/date-pickers/custom-field/CustomReadonlyBehaviorMaterialTextField.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { Dayjs } from 'dayjs';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers/DatePicker';
+import { DateFieldInPickerProps } from '@mui/x-date-pickers/DateField';
+import { useValidation, validateDate } from '@mui/x-date-pickers/validation';
+import { useSplitFieldProps, useFieldPlaceholder } from '@mui/x-date-pickers/hooks';
+import TextField from '@mui/material/TextField';
+
+function ReadonlyField(props: DateFieldInPickerProps<Dayjs, false>) {
```

Comment:
@LukasTy for this super basic example, I think the DX is pretty much as simple as I kind make it.

If you have ideas to improve it even further I'd be interested.
Otherwise I can next spend time trying to make sure that the complexity does not increase to much when you try to do more advanced stuff (the autocomplete, the masked editing, other editing strategy?).

Especially on the typing side, I think we reached a point where it is no longer an issue.

For the range demo, we need an equivalent of `DateFieldInPickerProps` but I want to advance on the multi vs single input fields before creating new interfaces.